WebRoot/js/zimbraMail/mail/model/ZmMailItem.js

```diff
@@ -176,6 +176,11 @@ function(node) {
 			contact = contactList && contactList.getContactByEmail(node.a),
 			fullName = contact && contact.getFullNameForDisplay(false);
 
+		// unless it's just the start of the email address (ZBUG-2236)
+		if (node.p != null && node.a != null && node.a.startsWith(fullName + "@")) {
+			fullName = node.p; // Update contacts with prettier name?
```

Comment:
Remove a comment.
FYI, "p" is personal name according to API reference.
https://files.zimbra.com/docs/soap_api/9.0.0/api-reference/zimbraMail/GetMsg.html

Reply:
Where are guidelines for this project on what makes a useful comment? I thought this one went along well with the one above it to explain the logic and give a ticket reference for context without searching the logs. Although I think it's useful, I don't have an issue with removing it. I just want to know what kind of comments are expected.

Reply:
We have internal wiki about coding style.
Regarding the PR, the comment at line 179 is useful as a reference. The comment at line 181 is unnecessary because it's clear enough from the code.
Additionally, we don't need to add a reference comment (like line 179) when the change is not complicated because we can find a commit id, a PR and a Jira ticket number using `git blame` command and blame view on github. Otherwise, we will add and see lots of reference comments. It is not good to keep the code readable.
In my opinion, we can remove the comment at 179 for the above reason.

Reply:
The first comment is there because the following code contradicts what the _existing_ comment at the beginning of the function above it says.

The second isn't describing that line, but more an XXX or TODO indicating that here's a place where we should maybe consider updating contacts with the prettier name we found. It wasn't describing the simple assignment. Maybe that should be entered as a separate ticket if necessary. For now, I just got rid of it.

Personally, I spend most of my time grepping and reading the source directly outside of GitHub or an IDE, so I appreciate a terse comment preventing me from having to dig through git blame or log then follow the ticket just to see why something was done. That is, I think they make the code more readable and friendly, as long as they're not entirely superfluous.  However, if the policy for the project is minimal comments, that's what I'll do.